### ISSUE TITLE
Null-values are not cacheable

### ISSUE THREAD

**Issue by ydewit** [ethlo/spring-tx-cache-decorator#6]
We have a service that returns null as a value, but it seems that EnhancedTransactionAwareCacheDecorator is treating that `explicitly set as deleted` in the code below:
```
    public ValueWrapper get(@NonNull final Object key)
    {
        final TransientCacheData tcd = tcd();
        final ValueWrapper res = tcd.getTransientCache().get(key);
        if (res != null && isNull(res))
        {
            // Was explicitly set as deleted
            return null;
        }
```
Returning Optional instead will also not work since Spring Cache unwraps it.

The end result is that we are seeing repeated calls to the underlying storage.

**Comment by ethlo** [comment]
Hi!

Would it not work to use a null-placeholder in the cache?

Something like

```java
        // Put somewhere static
        final Object nullObject = new Object();

        //mockTxnManager(true);
        //assertThat(TransactionSynchronizationManager.isSynchronizationActive()).isEqualTo(true);
        decoratorA.put("foo", nullObject);
        
        //assertThat(decoratorA.get("foo", Object.class)).isEqualTo(nullObject);
        
        //mockTransactionEnd(true);
        
        assertThat(realCacheA).containsEntry("foo", nullObject);
```

**Comment by ydewit** [comment]
That is the current workaround, but that means changing dozens of services. Ideally, I would expect the decorator to avoid changing the semantics of the caching provider.

**Comment by ethlo** [comment]
Ok, will look into what this may entail. What provider do you use?

**Comment by ydewit** [comment]
We use Ehcache.

**Comment by ethlo** [comment]
I have just released version 1.0.3 with the new fixes. Please confirm whether this fixes the issue.

**Comment by ydewit** [comment]
We had no time to validate, but I'll drop a note once we do.
